src/hooks/core/chat/useDMMessage.ts

```diff
@@ -51,7 +52,9 @@ const useDMMessage = () => {
         image: channelImage
       };
 
-      channel.firstMessage = channel?.messages?.[channel?.messages?.length - 1];
+      if (channel?.messages?.length > 0) {
+        channel.firstMessage = channel?.messages?.[channel?.messages?.length - 1];
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There's a potential issue with the assignment of `channel.firstMessage`. The code is assigning the last message in the array to `firstMessage`, which seems counterintuitive. If the intention is to assign the first message, you should use index 0 instead of `length - 1`. However, if the intention is indeed to assign the last message, consider renaming `firstMessage` to `lastMessage` for clarity.

```diff
- channel.firstMessage = channel?.messages?.[channel?.messages?.length - 1];
+ channel.lastMessage = channel?.messages?.[channel?.messages?.length - 1];
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->